ci-automation/vendor-testing/hetzner.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+# Copyright (c) 2023 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+# Test execution script for the Hetzner vendor.
+# This script is supposed to run in the mantle container.
+
+source ci-automation/vendor_test.sh
+
+hetzner_instance_type_var="HETZNER_${CIA_ARCH}_INSTANCE_TYPE"
+hetzner_instance_type="${!hetzner_instance_type_var}"
+
+# HETZNER_TPS_TOKEN should be provided by sdk_container/.env
+
+# We first need to create a temporary project using HETZNER_TPS_TOKEN
+# When the project is created it returns a regular HETZNER_TOKEN that can be used
+# in the next commands, it is a token similar to what you would get in your Hetzner console.
+HETZNER_TOKEN=$(curl \
+    --fail-with-body \
+    --retry 2 \
+    --silent \
+    --user-agent "flatcar-ci/unknown" \
```

Comment:
Could you add a comment about why we override the user agent?

Reply:
This looks like a debugging aid?

https://github.com/hetznercloud/tps-action/pull/5

Reply:
It works without the user agent. Having the user agent in requests is helpful for us operating the service as we can easily figure out who is affected by any errors, who is sending too many requests...

Of course, the user agent can easily be faked by anyone, so we don't fully rely on it